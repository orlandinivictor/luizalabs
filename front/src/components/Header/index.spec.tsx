import { describe, expect, it, vi } from "vitest";
import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";

import { Header } from ".";

describe("Header Tests", () => {
  it("match snapshot", () => {
    const navigate = vi.fn();
    const domTree = renderer.create(<Header navigate={navigate} />).toJSON();

    expect(domTree).toMatchSnapshot();
  });

  it("should be call navigate function", () => {
    const navigate = vi.fn();

    render(<Header navigate={navigate} />);
    fireEvent.click(screen.getByText("Wishlist"));

    expect(navigate).toBeCalled();
    expect(navigate).toBeCalledWith("/wishlist");
  });

  it("should call menu options function", async () => {
    const navigate = vi.fn();

    render(<Header navigate={navigate} />);
    fireEvent.click(screen.getByTestId("userButton"));

    await waitFor(() => screen.getByText("Entrar"));
    expect(screen.getByText("Entrar")).toBeDefined();
  });
});
