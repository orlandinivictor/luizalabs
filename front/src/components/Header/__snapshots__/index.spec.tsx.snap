// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Tests > match snapshot 1`] = `
[
  <div
    style={
      {
        "height": 0,
      }
    }
  />,
  <header
    className="sc-braxZu kfqJzj"
    showOptions={false}
  >
    <img
      alt="Logo Netshoes"
      src="/logo.svg"
    />
    <div>
      <button
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="Heart"
          >
            <path
              d="M12,20.043a.977.977,0,0,1-.7-.288L4.63,13.08A5.343,5.343,0,0,1,6.053,4.513,5.266,5.266,0,0,1,12,5.371a5.272,5.272,0,0,1,5.947-.858A5.343,5.343,0,0,1,19.37,13.08l-6.676,6.675A.977.977,0,0,1,12,20.043ZM8.355,4.963A4.015,4.015,0,0,0,6.511,5.4,4.4,4.4,0,0,0,4.122,8.643a4.345,4.345,0,0,0,1.215,3.73l6.675,6.675,6.651-6.675a4.345,4.345,0,0,0,1.215-3.73A4.4,4.4,0,0,0,17.489,5.4a4.338,4.338,0,0,0-4.968.852h0a.744.744,0,0,1-1.042,0A4.474,4.474,0,0,0,8.355,4.963Z"
            />
          </g>
        </svg>
        <span>
          Wishlist
        </span>
      </button>
      <div
        className="userContainer"
      >
        <button
          data-testid="userButton"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
        </button>
        <div
          className="userContent"
        >
          <div
            className="border"
          />
          <ul>
            <li>
              <button
                onClick={[Function]}
              >
                Entrar
              </button>
            </li>
            <li>
              <button
                onClick={[Function]}
              >
                Minha Conta
              </button>
            </li>
            <li>
              <button
                onClick={[Function]}
              >
                Endere√ßos
              </button>
            </li>
            <li>
              <button
                onClick={[Function]}
              >
                Minha Netshoes
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>,
]
`;
